### PR TITLE
ICU-21052 Fix two compilations with UCONFIG_NO_CONVERSION option in t…

### PR DESCRIPTION
**Pull request by sven-oly**
…ools.toolutil

##### Checklist

- [x] Issue filed: https://unicode-org.atlassian.net/browse/ICU-21052
- [x] Updated PR title and link in previous line to include Issue number
- [x] Issue accepted
- [x] Tests included
- [ ] Documentation is changed or added

### CODE REVIEW

**Comment by sven-oly** [comment]
Note that this does not fix compilation of icu4c/source/tools/tooluti, but only these two files. There is no requirement or expectation that all tools will compile with UCONFIG_NO_CONVERSION set to 1.

File: icu4c/source/tools/toolutil/dbgutil.cpp [hunk 1]
```diff
@@ -29,7 +29,7 @@ static const UnicodeString&  _fieldString(UDebugEnumType type, int32_t field, Un
     if(str == NULL) {
         return fillin.remove();
     } else {
-        return fillin = UnicodeString(str, ""); // optimize?
+      return fillin = UnicodeString(str, -1, US_INV);
```
Comment (sffc) [inline icu4c/source/tools/toolutil/dbgutil.cpp:32]: Why did you change the indentation?

Comment (sven-oly) [reply icu4c/source/tools/toolutil/dbgutil.cpp]: Ah, didn't intend to. Updated.
